Realm/Tests/DynamicTests.m

```diff
@@ -161,7 +161,7 @@ - (void)testDynamicTypes {
 
     // verify properties
     RLMRealm *dyrealm = [self realmWithTestPathAndSchema:nil];
-    RLMResults *results = [dyrealm allObjects:AllTypesObject.className];
+    RLMResults<RLMObject *> *results = [dyrealm allObjects:AllTypesObject.className];
```

Comment:
All of the tests build and pass with no changes to the tests other than updating what error message a few of the tests are looking for.